src/recipe/parser/glob_vec.rs

```diff
@@ -0,0 +1,140 @@
+use globset::{Glob, GlobSet};
+
+use serde::ser::SerializeSeq;
+use serde::{Deserialize, Serialize};
+
+use crate::_partialerror;
+use crate::recipe::custom_yaml::{HasSpan, RenderedNode, RenderedSequenceNode, TryConvertNode};
+use crate::recipe::error::{ErrorKind, PartialParsingError};
+
+#[derive(Default, Debug, Clone)]
+pub struct GlobVec(Vec<Glob>, Option<GlobSet>);
+
+impl Serialize for GlobVec {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
+        for glob in self.0.iter() {
+            seq.serialize_element(glob.glob())?;
+        }
+        seq.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for GlobVec {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut vec = Vec::deserialize(deserializer)?;
+        let mut glob_vec = Vec::with_capacity(vec.len());
+        for glob in vec.drain(..) {
+            let glob = Glob::new(glob).map_err(serde::de::Error::custom)?;
+            glob_vec.push(glob);
+        }
+
+        if glob_vec.is_empty() {
+            Ok(Self(glob_vec, None))
+        } else {
+            let mut globset_builder = globset::GlobSetBuilder::new();
+            for glob in glob_vec.iter() {
+                globset_builder.add(glob.clone());
+            }
+            let globset = globset_builder.build().map_err(serde::de::Error::custom)?;
+
+            Ok(Self(glob_vec, Some(globset)))
+        }
+    }
+}
+
+impl GlobVec {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn globset(&self) -> Option<&GlobSet> {
```

Comment:
Should we also add a getter method for getting the underlying Vector of `GlobMatcher`s? Or is `GlobSet` enough to perform matches?

Reply:
Let's do that when we need it? Maybe `globset` is enough -- that depends I'd say.

Reply:
Alright! I will switch to using `GlobVec` after this PR is merged and see if that's necessary.